apis/core/v1alpha1/clusterobjectdeployment_types.go

```diff
@@ -0,0 +1,59 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ClusterObjectDeploymentSpec defines the desired state of a ClusterObjectDeployment.
+type ClusterObjectDeploymentSpec struct {
+	// Number of old revisions in the form of archived ObjectSets to keep.
+	// +kubebuilder:default=5
+	RevisionHistoryLimit *int `json:"revisionHistoryLimit,omitempty"`
```

Comment:
Kube is using `*int32` and default `10`, maybe we should do the same?
https://github.com/kubernetes/api/blob/master/apps/v1/types.go#L370-L374

Please also change in ObjectDeploymentSpec!